tests/unit/test_plugin.py

```diff
@@ -0,0 +1,113 @@
+#
+# Copyright 2023 SUSE LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import datetime
+import pytest
+import urllib.error
+
+from unittest.mock import Mock, patch
+
+from csp_billing_adapter_microsoft import plugin
+from csp_billing_adapter.config import Config
+from csp_billing_adapter.adapter import get_plugin_manager
+import csp_billing_adapter.exceptions as cba_exceptions
+
+
+pm = get_plugin_manager()
+config = Config.load_from_file(
+    'tests/data/good_config.yaml',
+    pm.hook
+)
+
+
+@patch('csp_billing_adapter_microsoft.plugin._is_required_metadata_version_available')
+def test_setup(mock_check_metadata_version):
+    mock_check_metadata_version is True
+    plugin.setup_adapter(config)  # Currently no-op
+
+# @patch('csp_billing_adapter_microsoft.plugin._is_required_metadata_version_available')
+# def test_setup_adapter_fails(mock_check_metadata_version):
+#     mock_check_metadata_version is False
+#     with pytest.raises(cba_exceptions.MetadataCollectorError):
+#         plugin.setup_adapter(config)  # Cur
+
+
+def test_meter_billing():
+    pass
+
+
+def test_meter_billing_error():
+    pass
+
+
+def test_get_csp_name():
+    assert plugin.get_csp_name(config) == 'microsoft'
+
+
+@patch('csp_billing_adapter_microsoft.plugin.urllib.request.urlopen')
+def test_get_account_info(mock_urlopen):
+    urlopen = Mock()
+    urlopen.read.side_effect = [
+        b'{"compute": "info", "network": "info"}',
+        b'{"signature": "signature", "pkcs7": "pkcs7"}'
+        # b'signature',
```

Comment:
These can be cleaned up?

Reply:
Thanks @smarlowucf for catching this. I removed the extra lines.